oe_theme.theme

```diff
@@ -1234,3 +1234,59 @@ function oe_theme_preprocess_pattern_link__preview(array &$variables): void {
   // Pattern preview will pass along a \Drupal\Core\Render\Markup object.
   $variables['url'] = Url::fromUserInput((string) $variables['url']);
 }
+
+/**
+ * Implements hook_preprocess_paragraph() for oe_banner paragraph.
+ */
+function oe_theme_preprocess_paragraph__oe_banner(array &$variables): void {
+  /** @var \Drupal\paragraphs\Entity\Paragraph $paragraph */
+  $paragraph = $variables['paragraph'];
+
+  $variant = $paragraph->get('oe_paragraphs_variant')->first()->value;
+  $variables['variant'] = $variant ?? 'default';
```

Comment:
Unfortunately, the variant names that come from the paragraph don't match the variants you defined. The variants that come from the paragraphs are the ids of the form view modes (oe_banner_image, oe_banner_image_shade and oe_banner_primary), so you will need to process them here or rename them on the paragraphs PR.